components/dotenv.rst

```diff
@@ -67,6 +67,42 @@ The ``load()`` method never overwrites existing environment variables. Use the
 .. versionadded:: 4.2
     The ``Dotenv::overload()`` method was introduced in Symfony 4.2.
 
+Symfony follows and promotes the `app config recommendations`_ made by the
+Twelve-Factor App methodology. Among other things, it's recommended to not group
+env vars into environments (``dev``, ``prod``, ``test``, etc.)
+
+When deploying apps you should follow that recommendation strictly. However,
+when developing apps on your local machine, testing multiple apps in different
+environments with regular env vars is cumbersome. That's why Symfony provides a
+feature to define env vars per environment using the ``loadForEnv()`` method
+(but you should only use it on your local machine, not in production servers)::
+
+    // ...
+    $dotenv->loadForEnv('dev', __DIR__.'/.env');
+    // Looks for env vars in these files (and in this order):
+    //   .env.dev.local
+    //   .env.local
```

Comment:
Should be loaded **after** `.env.dev`

Reply:
> Will be loaded after .env.dev

Hmm. Are you sure? I didn't check the code, but if it's true it's not respecting the same priorities as https://github.com/bkeepers/dotenv/blob/master/README.md#what-other-env-files-can-i-use, i.e:

- `.env.[env].local`
- `.env.local`
- `.env.[env]`
- `.env`

Also, what about using the same presentation as `bkeepers/dotenv`? (The table with complementary information)

Reply:
But https://facebook.github.io/create-react-app/docs/adding-custom-environment-variables#what-other-env-files-can-be-used does load `.env.local` after `.env.[env]`...

IMHO, loading `.env.local` makes more sense, as `local` should always allow env override. Either you use `.env.local` to override for all environments, either you use `.env.[env].local` for a specific env. But `.env` and `.env.[env]` will only contains sensible defaults, so `.env.[env]` should not have priority over `.env.local`.